Competition2017/src/competition/subsystems/agitator/AgitatorSubsystem.java

```diff
@@ -41,30 +59,62 @@ public AgitatorSubsystem(
         this.agitatorMotor.createTelemetryProperties(side.toString(), propManager);
     }
 
-    public void setAgitatorPower(double power) {
+    protected void setAgitatorPowerRaw(double power) {
         agitatorMotor.ensureTalonControlMode(TalonControlMode.PercentVbus);
         agitatorMotor.set(power);
     }
+    
+    public void setAgitatorPower(double power) {
+        setAgitatorPowerRaw(power);
+        unjamEnd = -1;
+        isIntaking = false;
+    }
   
     public RobotSide getSide() {
         return side;
     }
 
     public void eject() {
-        agitatorMotor.set(ejectPowerProperty.get());
+        setAgitatorPower(ejectPowerProperty.get());
     }
 
     public void intake() {
-        agitatorMotor.set(intakePowerProperty.get());
+        if(unjamEnd < 0 || Timer.getFPGATimestamp() > unjamEnd) {
+            setAgitatorPowerRaw(intakePowerProperty.get());
+        }
+        isIntaking = true;
     }
 
     public void stop() {
-        agitatorMotor.set(0);
+        setAgitatorPower(0);
     }
 
     @Override
     public void updatePeriodicData() {
         agitatorMotor.updateTelemetryProperties();
+        if(agitatorMotor.getOutputCurrent() > agitatorOverCurrentThreshold.get()) {
```

Comment:
consider logging a latch about the different state changes here (over current, over time, unjamming, unjamming end etc..)